client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-reprint-modal.js

```diff
@@ -30,23 +31,26 @@ const ReprintDialog = ( props ) => {
 			action: 'confirm',
 			onClick: onConfirm,
 			isPrimary: true,
-			disabled: reprintDialog && reprintDialog.isFetching,
-			additionalClassNames: reprintDialog && reprintDialog.isFetching ? 'is-busy' : '',
-			label: translate( 'Print' ),
+			disabled: isFetching,
+			additionalClassNames: isFetching ? 'is-busy' : '',
+			label: download ? translate( 'Download' ) : translate( 'Print' ),
 		},
 	];
 
 	return (
 		<Dialog
-			isVisible={ Boolean( reprintDialog && reprintDialog.labelId === labelId ) }
+			isVisible={ reprintLabelId === labelId }
 			onClose={ onClose }
 			buttons={ buttons }
 			additionalClassNames="label-reprint-modal woocommerce wcc-root">
 			<FormSectionHeading>
-				{ translate( 'Reprint shipping label' ) }
+				{ download ? translate( 'Redownload shipping label' ) : translate( 'Reprint shipping label' ) }
```

Comment:
Same here about the "Redownload". If it sounds good to a native Engligh speaker, then I'm fine, but it certainly sounds weird to me...